### DISCUSSION TITLE
Is it easy to port syntax color schemes from lite-xl?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#375] [Q&A] [answered]
I have read this section of README.md, but I don't know how to start yet.

https://github.com/SpartanJ/ecode?tab=readme-ov-file#customizations

Where are the existing syntax color schemes stored? I think I will try to modify an existing syntax color scheme first.

**Comment by SpartanJ** [comment, marked as answer]
It's kinda trivial but there's no script provided to convert them automatically given that ecode has its own properties that lite-xl might not have. But original lite color names are the same used in ecode so the majority of the properties are the same:

Color schemes in ecode are located [here](https://github.com/SpartanJ/eepp/blob/develop/bin/assets/colorschemes/colorschemes.conf) and you can add more following the link you provided.

For direct comparison you can take a look at any existing editor color scheme, let's use Dracula:

ecode:
```ini
[dracula]
background = #282a36
text = #7b81a6
caret = #f8f8f0
accent = #8be9fd
selection = #44475a
line_number = #53576e
line_number2 = #f8f8f0
line_highlight = #313442

normal = #f8f8f2
symbol = #f8f8f2
comment = #6272a4
keyword = #ff79c6
keyword2 = #8be9fd
keyword3 = #ffb86c
number = #bd93f9
literal = #f1fa8c
string = #f1fa8c
operator = #bd93f9
function = #50fa7b
```

lite-xl:
```lua
local style = require "core.style"
local common = require "core.common"

style.background = { common.color "#282a36" }
style.background2 = { common.color "#21222b" }
style.background3 = { common.color "#21222b" }
style.text = { common.color "#7b81a6" }
style.caret = { common.color "#f8f8f0" }
style.accent = { common.color "#8be9fd" }
style.dim = { common.color "#4f5873" }
style.divider = { common.color "#1f2029" }
style.selection = { common.color "#44475a" }
style.line_number = { common.color "#53576e" }
style.line_number2 = { common.color "#f8f8f0" }
style.line_highlight = { common.color "#313442" }
style.scrollbar = { common.color "#44475a" }
style.scrollbar2 = { common.color "#ff79c6" }

style.syntax["normal"] = { common.color "#f8f8f2" }
style.syntax["symbol"] = { common.color "#f8f8f2" }
style.syntax["comment"] = { common.color "#6272a4" }
style.syntax["keyword"] = { common.color "#ff79c6" }
style.syntax["keyword2"] = { common.color "#ff79c6" }
style.syntax["number"] = { common.color "#bd93f9" }
style.syntax["literal"] = { common.color "#f1fa8c" }
style.syntax["string"] = { common.color "#f1fa8c" }
style.syntax["operator"] = { common.color "#ff79c6" }
style.syntax["function"] = { common.color "#50fa7b" }
```

It's self explanatory, matching names can be used.

Since ecode adds some more properties if those are not defined it will fallback to using one of the base colors. The complete list of properties is in the base color scheme [eepp](https://github.com/SpartanJ/eepp/blob/develop/bin/assets/colorschemes/colorschemes.conf#L1). ecode also can define background color (second color separeted by comma), and font style: shadow, underline, strikethrough.
